### PR TITLE
Fix flaky tests based on encoded html entities

### PR DESCRIPTION
**Pull request by ruslankhaertdinov**
Issue: When `'` becomes `&#39;` on text comparisons:
```ruby
expected the body to contain "Jermain O'Keefe" but was "<p>Jermain O&#39;Keefe!</p>
```
Solution: using another `Faker`'s method, which generates string without special chars.

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FactoryGirl.define do
   factory :user do
     email
-    full_name { Faker::Name.name }
+    full_name { Faker::Name.name.gsub("'", "") }
```
Comment (timurvafin) [inline spec/factories/users.rb]: @ruslankhaertdinov could it be done in the Faker in self? B/c/ there are more cases like `&`

Comment (ruslankhaertdinov) [inline spec/factories/users.rb]: @timurvafin Thanks, I'll research

**Comment by ruslankhaertdinov** [comment]
@timurvafin seems like words (`first_name`, `last_name` etc.) accessed directly from Faker's `locale` file. I see 2 ways to solve problem: 
1) Override parts with inappropriate content in local `locale` file
2) Create helper method for sanitizing strings. Something like this:
```ruby
def sanitize_string(string)
  string.gsub(/[^\w\s]/, "")
end
```
And sanitize generated string:
```ruby
factory :user do
  full_name { sanitize_string(Faker::Name.name) }
...
```
What do you think about this approaches?

**Comment by timurvafin** [comment]
@ruslankhaertdinov could found code which converts `'` to `&#39;` and use it somehow in `sanitize_string`?

**Comment by ruslankhaertdinov** [comment]
@timurvafin what about if in specs we will compare escaped html strings (example in current PR)?

**Comment by KirillKayumov** [comment]
@ruslankhaertdinov I don't think that including `ERB::Util` and using `#html_escape` method in every feature spec where we may have a deal with special chars like `'` is a good solution. 

Also using special methods like `#sanitize_string` in factories which encode special chars to html entities is not a good option. For example our new created user has name `D'angelo` and if we want to check it, we should type `expect(user.name).to eq("D&#39;angelo")`. Looks oddly.

I suggest to simply use other method from Faker. For example `Faker::Internet.user_name` or even `Faker::Name.title`. These methods return string without special chars. Eventually we can specify constant string for user name and do not use Faker here.

CC @timurvafin

**Comment by ruslankhaertdinov** [comment]
@KirillKayumov thanks for suggestion :+1:  I'm considering various options and looking for more convenient and easy

**Comment by ruslankhaertdinov** [comment]
@KirillKayumov please take a look again)

**Comment by KirillKayumov** [comment]
Looks :ok: for me

**Comment by IlkhamGaysin** [comment]
If we use further email spec together with faker, then it is always necessary to remember what won't pass a test. 
Perhaps it is worth testing mailers without the aid of email spec and separately from feature of tests.?
It is also possible to rewrite the devise mailers test on similarity of the following:

```
require "rails_helper"

feature DeviseMailer do
  let(:user) { create :user }
  let(:support_email) { ENV["MAILER_SENDER_ADDRESS"] }
  let(:subject) { "Confirmation instructions" }

  feature "#confirmation" do
    let(:mail) { send_confirmation }

    scenario "match mailer" do
      expect(mail.subject).to eq(subject)
      expect(mail.to).to eq([user.email])
      expect(mail.from).to eq([support_email])
      expect(mail.body.decoded).to have_content(user.full_name)
    end
  end

  private

  def send_confirmation
    user.send(:generate_confirmation_token)
    described_class.confirmation_instructions(user, user.instance_variable_get(:@raw_confirmation_token))
  end
end

```

**Comment by timurvafin** [comment]
@ruslankhaertdinov @IlkhamGaysin please reopen when will be updated.

**Pull request closed without merging**